src/templates/antora/util.ts

```diff
@@ -0,0 +1,270 @@
+
+interface NavFile {
+  readonly title: string;
+  readonly path: string;
+  readonly pages?: NavFile[];
+}
+
+export interface PageOutput {
+  readonly type: 'adoc' | 'json';
+  readonly filename: string;
+  readonly content: string;
+}
+
+const BASE_PATH: string = process.env.BASE_PATH || '/_data/antora';
+
+const namespaceDescriptions = {
+  'tinymce': 'Global APIs for working with the editor.',
+  'tinymce.dom': 'APIs for working with the DOM from within the editor.',
+  'tinymce.editor.ui': 'APIs for registering User Interface components.',
+  'tinymce.geom': 'Various rectangle APIs.',
+  'tinymce.html': 'APIs for working with HTML within the editor.',
+  'tinymce.util': 'Browser related APIs.'
+};
+
+/**
+ * [compileTemplate description]
+ * @param  {[type]} filePath [description]
+ * @return {[type]}          [description]
+ */
+const compileTemplate = (filePath: string): HandlebarsTemplateDelegate => {
+  return Handlebars.compile(fs.readFileSync(path.join(__dirname, filePath)).toString());
+};
+
+const getNamespaceFromFullName = (fullName: string): string =>
+  fullName === 'tinymce' ? fullName : fullName.split('.').slice(0, -1).join('.');
+
+const getApiFromFullName = (fullName: string): string =>
+  fullName === 'tinymce' ? fullName : fullName.split('.').slice(1).join('.');
+
+const getNameFromFullName = (fullName: string): string =>
+  fullName === 'tinymce' ? fullName : fullName.split('.').slice(-1).join('.');
+
+const getTitleFromFullName = (fullName: string): string =>
+  fullName.split('.').slice(-1).join('');
+
+const navLine = (name: string, level: number): string =>
+  '*'.repeat(level) + ' ' + name + '\n';
+
+const generateNavXref = (basePath: string, filename: string, title: string): string =>
+  'xref:' + basePath + '/' + filename + '[' + title + ']';
+
+const generateXref = (name: string, structure: ExportStructure): string => {
+  const title = getTitleFromFullName(name);
+  const fileName = name.toLowerCase() === 'tinymce' ? getRootPath(structure) : name.toLowerCase();
+  switch (structure) {
+    case 'legacy':
+      return generateNavXref('api/' + getNamespaceFromFullName(name.toLowerCase()), fileName + '.adoc', title);
+
+    case 'default':
+      return generateNavXref('apis', fileName + '.adoc', title);
+  }
+};
+
+const generateTypeXref = (type: string, structure: ExportStructure): string => {
+  return type.includes('tinymce', 0) ? generateXref(type, structure) : type;
+};
+
+const getJsonFilePath = (type: string, fullName: string): string =>
+  ('_data/api/json/' + type + '_' + fullName.replace(/\./g, '_') + '.json').toLowerCase();
+
+const getFilePath = (name: string, structure: ExportStructure): string => {
+  const fileName = name.toLowerCase() === 'tinymce' ? getRootPath(structure) : name.toLowerCase();
+  switch (structure) {
+    case 'legacy':
+      const folder = getNamespaceFromFullName(name) + '/';
+      return BASE_PATH + '/api/' + folder + '/' + fileName + '.adoc';
```

Comment:
I think the bug is here. You could have a function that doesn't care about extra slashes and just normalizes them something like:
```
const joinPaths = (paths: string[]): string => paths.flatMap((part) => part.split('/')).filter((p) => p.length > 0).join('/');
```
Or maybe you could use node js Path.join function to make this a bit more safe. Unsure though how that acts on other platforms. Seems to be doing special things on win32 https://github.com/nodejs/node/blob/main/lib/path.js#L484

Reply:
The issue would definitely be the additional slash added to the `folder`, I'm fixing this one and updating each usage of variables in strings to use jquery syntax.

Reply:
Should be easier to spot these things not with the verbose reporting we could even enable that in CI so make sure that it doesn't do things like this in the future.